ext/posix/tests/posix_isatty_stream_lose_data.phpt

```diff
@@ -0,0 +1,33 @@
+--TEST--
+posix_isatty(): casting stream emits data loss
+--EXTENSIONS--
+posix
+--SKIPIF--
+<?php
+//if (getenv('SKIP_ONLINE_TESTS')) die('skip online test');
+require __DIR__ . '/../../standard/tests/http/server.inc'; http_server_skipif();
+?>
+--INI--
+allow_url_fopen=1
+--FILE--
+<?php
+
+require __DIR__ . '/../../standard/tests/http/server.inc';
+
+$responses = array(
+    "data://text/plain,HTTP/1.0 200 Ok\r\nSome: Header\r\nSome: Header\r\n\r\nBody",
+);
+
+['pid' => $pid, 'uri' => $uri] = http_server($responses, $output);
+
+$handle = fopen($uri, 'r');
```

Comment:
It should be probably pointed out that this is a bogus case - there is no data loss in reality and just the warning is wrong. Although it's still a useful test as it tests the actual cast so it might make sense to leave and possibly just repurpose in the future.

It might be still worth to check if there is some other case where data loss is genuine. In such case this PR would be a bug fix. Otherwise it is a feature and should go only to master.

Reply:
The issue is that I haven't been able to reproduce a case where the data loss is actually real.
I don't mind to backport this just as a precaution but I haven't done a deep dive in how streams work yet.

Reply:
Data loss could happen if we replaced `posix_isatty($handle)` in this test by some other function that casts _and reads_ from $handle. See https://github.com/php/php-src/pull/10093#issuecomment-1373485178